packages/docusaurus-theme-classic/src/routeAnnouncer.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import {translate} from '@docusaurus/Translate';
+import type {ClientModule} from '@docusaurus/types';
+import './routeAnnouncer.css';
+
+const clientModule: ClientModule = {
+  onRouteDidUpdate({location, previousLocation}) {
+    if (!previousLocation || location.pathname === previousLocation.pathname) {
+      return;
+    }
+    Array.from(
+      document.getElementsByTagName('docusaurus-route-announcer'),
+    ).forEach(document.removeChild);
+    const announcerContainer = document.createElement(
+      'docusaurus-route-announcer',
+    );
+    const announcement = document.createElement('p');
+    Object.assign(announcement, {
+      ariaLive: 'assertive', // Make the announcement immediately.
+      id: 'docusaurus-route-announcer-content',
+      role: 'alert',
+    });
```

Comment:
Pure DOM manipulation. Because client modules don't have access to the React context, it would be quite cumbersome if we want to use React to do this. The alternative will be to put this in `Layout`, but there may be other edge-cases (for example, the layout doesn't re-mount between doc pages)